queue/config.go

```diff
@@ -0,0 +1,38 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package queue
+
+import (
+	"fmt"
+	"time"
+
+	diskqueue "github.com/elastic/beats/v7/libbeat/publisher/queue/diskqueue"
+	memqueue "github.com/elastic/beats/v7/libbeat/publisher/queue/memqueue"
+)
+
+type Config struct {
+	Type         string             `config:"type"`
+	MemSettings  memqueue.Settings  `config:"memqueue"`
```

Comment:
Is there a way to make these pointers so we don't need the type parameter? You can just infer it based on whether you have mem settings or queue settings?

Reply:
switched to pointers